.github/workflows/node-ci.yml

```
@@ -228,6 +229,111 @@ jobs:
             # Do not place any steps after "Check Builds"
             #
 
+    check_schema:
+        name: Check for schema changes (answerful and answerless)
+        runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-latest]
+                node-version: [20.x]
+        steps:
+            - name: Check out latest commit
+              uses: actions/checkout@v4
+
+            - name: Install & cache node_modules
+              uses: ./.github/actions/shared-node-cache
+              with:
+                  node-version: ${{ matrix.node-version }}
+
+            - name: Build schema for current commit
+              run: >-
+                  pnpm tsc --declaration --emitDeclarationOnly \
+                    --outfile ${{ runner.temp }}/schema-pr.d.ts \
+                    packages/perseus-core/src/data-schema.ts
+                  pnpm prettier --write ${{ runner.temp }}/schema-pr.d.ts
+
+            - name: Check out base branch
+              uses: actions/checkout@v4
+              with:
+                  # This checks out the base branch of the PR (the branch the PR
+                  # is targeting)
+                  ref: ${{ github.event.pull_request.base.ref }}
+
+            - name: Re-install node_modules for base branch
+              run: pnpm install
+
+            - name: Build schema for base
+              run: >-
+                  pnpm tsc --declaration --emitDeclarationOnly \
+                    --outfile ${{ runner.temp }}/schema-base.d.ts \
+                    packages/perseus-core/src/data-schema.ts
+
+                  pnpm prettier --write ${{ runner.temp }}/schema-base.d.ts
+
+            - name: Check for Schema Change
+              id: check_for_schema_change
+              run: |
+                  # So a non-zero exit code from 'diff' doesn't bail this step.
+                  set +e
+                  # We generate a random DELIMeter so that the existence of the
```

Reading this top-to-bottom, this was confusing. I thought you were making some kind of deli joke.

```suggestion
                  # We generate a random delimiter (DELIM) so that the existence of the
```